Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+
+#include <AZTestShared/Math/MathTestHelpers.h>
+#include <AZTestShared/Utils/Utils.h>
+
+#include <PhysXCharacters/API/CharacterController.h>
+#include <PhysXCharacters/API/CharacterUtils.h>
+#include <PhysXCharacters/Components/CharacterControllerComponent.h>
+#include <PhysXCharacters/Components/CharacterGameplayComponent.h>
+
+#include <PhysX/ComponentTypeIds.h>
+#include <PhysX/SystemComponentBus.h>
+
+#include <Source/CapsuleColliderComponent.h>
+#include <Source/SphereColliderComponent.h>
+
+#include <Tests/PhysXTestCommon.h>
+#include <Tests/PhysXTestFixtures.h>
+#include <Tests/PhysXTestUtil.h>
+
+#include <System/PhysXSystem.h>
+
+namespace PhysX
+{
+    // transform for a floor centred at x = 0, y = 0, with top at level z = 0
+    static const AZ::Transform DefaultFloorTransform = AZ::Transform::CreateTranslation(AZ::Vector3::CreateAxisZ(-0.5f));
+
+    // Default Values for the Gamplay Controller
+    static const float DefaultGravityMultiplier = 1.0f;
+    static const float DefaultGroundDetectionBoxHeight = 0.05f;
+
+    class ControllerTestBasis
```

Comment:
This class seem repeated in `CharacterControllerTests.cpp`, it can be moved to a .h file so both of the cpp files include it.

Reply:
@moraaar this isn't the same class. It's similar in that it stands up a transform component and character controller, but it also stands up a gameplay component and uses a different pattern for the setup due to the needs and usage. Lastly, the update is a different behavior as well.

I will change the name of this test basis class to help drive that it's not the same class as the `ControllerTestBasis` found in the Character Controller Tests.

Reply:
Changing the name of the class is what I would do as well, thanks!